doc/faascontroller.md

```diff
@@ -0,0 +1,206 @@
+image:          "dev.local/colordeploy:17.0"    
+port:           8089
+autoScaleType:  "rps"          
+autoScaleValue: "111"          
+minReplica:     1
+maxReplica:     3
+limitCPU:       "180m"         
+limitMemory:    "100Mi"        
+requestCPU:     "80m"          
+requestMemory:  "20Mi"         
+requestAdaptor:                       
+  header:
+    set:
+      X-Func1: func-demo-10              # add one HTTP header  
+```
+
+### Lifecycle
+There four types of function status, Pending, Active, InActive, and Failed[4]. Basically, they come from AWS Lambda's status.
+* Once the function has been created in Easegress, its original status is `pending`. After the function had been provisioned successfully by FaaSProvider(Knative), its status will become `active`. At last, FaaSController will add the routing rule in its HTTPServer for this function.  
+* Easegress's FaaSFunction will be `active` not matter there are requests or not. Stop function execution by calling FaaSController's `stop` RESTful API, then it will turn function into `inactive`.  Updating function's spec for image URL or else fields, or deleting function also need to stop it first.
+* **Easegress will only route ingress traffic to FaaSProvider when the function is in the `active` state.**
+* After updating the function, it will run into `pending` status. (Once provision successfully, it will become `active` automatically) 
+* FaaSController will remove the routing rule in its HTTPServer for rejecting one function's ingress traffic when it's in `inactive` status. (Then the client will receive HTTP 404 failure response. For becoming zero-downtime, deploy another new FaaSFunction in this FaaSController may be helpful)
+
+```                                                      
+                                                         +-------------------+
+                                                         | provision succ    |
+  +---------------+   provision succ              +------+------+<-----------+
+  |               |------------------------------>|             |
+  |    pending    |             provision pending |   active    |<------+
+  |               | <-----------------------------|             |       |
++-+---------------+ <--+     +--------------------+--------+----+       |   
+|                      |     |                             |            |
+provision faild        |     |                             |stop        | start
+|                  update    provision failed              |            |
+|  +---------------+   |     |        +-------------+      |            |
++->|               |   |     |        |             |      |            |
+   |   failed      |<--+-----+--------+   inactive  |<-----+            |
+   |               |                  |             +-------------------+
+   |               |                  |             |
+   +---------------+                  +-------------+
```

Comment:
```suggestion
                  Provision

                      │
                      │
                      │
                ┌─────▼──────┐      Start     ┌────────────┐
                │            │     Success    │            │
       ┌────────┤  Pending   ├────────────────►   Active   │
       │        │            │                │            ├──────┐
       │        └───┬───▲────┘                └────┬───▲───┘      │
       │            │   │                          │   │          │
       │            │   │                          │   │          │
       │      Start │   │ Update             Stop  │   │ Start    │
       │     Failed │   │                          │   │Success   │
       │            │   │                          │   │          │
       │        ┌───▼───┴────┐                ┌────▼───┴───┐      │
       │        │            │                │            │      │
Delete │        │   Failed   ◄────────────────┤  Inactive  │      │
       │        │            │  Start Failed  │            │      │
       │        └───┬───▲────┘                └──────┬─────┘      │
       │            │   │                            │            │
       │            │   │              Errors        │            │
       │    Delete  │   └────────────────────────────┼────────────┘
       │            │                                │
       │            │                                │
       │        ┌───▼────────┐                       │
       │        │            │          Delete       │
       └────────►  Destory   ◄───────────────────────┘
                │            │
                └────────────┘
```

Reply:
The original state machine is not right. I proposed this one.

Reply:
All related suggestions have been updated manually. Thx for correcting.